osu.Game/Overlays/Login/SecondFactorAuthForm.cs

```diff
@@ -0,0 +1,117 @@
+                    AutoSizeAxes = Axes.Y,
+                    Padding = new MarginPadding { Horizontal = SettingsPanel.CONTENT_MARGINS },
+                    Direction = FillDirection.Vertical,
+                    Spacing = new Vector2(0f, SettingsSection.ITEM_SPACING),
+                    Children = new Drawable[]
+                    {
+                        new OsuTextFlowContainer(s => s.Font = OsuFont.GetFont(weight: FontWeight.Regular))
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Text = "An email has been sent to you with a verification code. Enter the code.",
+                        },
+                        codeTextBox = new OsuTextBox
+                        {
+                            PlaceholderText = "Enter code",
+                            RelativeSizeAxes = Axes.X,
+                            TabbableContentContainer = this,
+                        },
+                        explainText = new LinkFlowContainer(s => s.Font = OsuFont.GetFont(weight: FontWeight.Regular))
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                        },
+                        errorText = new ErrorTextFlowContainer
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Alpha = 0,
+                        },
+                    },
+                },
+                new LinkFlowContainer
+                {
+                    Padding = new MarginPadding { Horizontal = SettingsPanel.CONTENT_MARGINS },
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                },
+            };
+
+            explainText.AddParagraph(UserVerificationStrings.BoxInfoCheckSpam);
+            // We can't support localisable strings with nested links yet. Not sure if we even can (probably need to allow markdown link formatting or something).
+            explainText.AddParagraph("If you can't access your email or have forgotten what you used, please follow the ");
+            explainText.AddLink(UserVerificationStrings.BoxInfoRecoverLink, $"{api.WebsiteRootUrl}/home/password-reset");
+            explainText.AddText(". You can also ");
+            explainText.AddLink(UserVerificationStrings.BoxInfoReissueLink, () =>
+            {
+                // TODO: request another code.
+            });
+            explainText.AddText(" or ");
+            explainText.AddLink(UserVerificationStrings.BoxInfoLogoutLink, () => { api.Logout(); });
+            explainText.AddText(".");
+
+            codeTextBox.Current.BindValueChanged(code =>
+            {
+                if (code.NewValue.Length == 8)
+                {
+                    api.AuthenticateSecondFactor(code.NewValue);
```

Comment:
This is presumably going to be a web request wherein the client sends the second factor code to web and validates it. If my thinking there is correct, this is gonna need a loading spinner for when the 2FA code is verified.

Reply:
It already changes the api state to "Connecting" again while performing the 2FA (you can test this, I added a sleep for testing purposes while in draft; you can also test success / failure as the "valid" code is all zeroes).